### ISSUE TITLE
Gravity 7.0.22/23 do not install when invoked outside the unpacked tarball

### ISSUE THREAD

**Issue by wadells** [gravitational/gravity#2292]
### Description

**What happened**:

I tried to run a 7.0.23 gravity install from a directory that wasn't the unpacked installer tarball directory.

**What you expected to happen**:
I expected gravity to install, as it does in 7.0.16.

**How to reproduce it (as minimally and precisely as possible)**:

Download & unpack 7.0.16 and 7.0.23 tarballs. Move to a directory that isn't an unpacked installer tarball.

```
-bash-4.2$ ls ~/
7.0.16  7.0.23  test
-bash-4.2$ pwd
/home/centos/test
```

Try to install 7.0.16:

```
-bash-4.2$ sudo /home/centos/7.0.16/gravity install --cloud-provider=generic                    
Thu Oct 29 00:21:04 UTC Starting enterprise installer                
                                                                                               
To abort the installation and clean up the system,
press Ctrl+C two times in a row.                                                               
                                                                                               
If you get disconnected from the terminal, you can reconnect to the installer
agent by issuing 'gravity resume' command.                                                     
                                               
If the installation fails, use 'gravity plan' to inspect the state and
'gravity resume' to continue the operation.                                                    
See https://gravitational.com/gravity/docs/cluster/#managing-an-ongoing-operation for details.
                                               
Thu Oct 29 00:21:04 UTC Connecting to installer
Thu Oct 29 00:21:18 UTC Connected to installer
Thu Oct 29 00:21:18 UTC Successfully added "node" node on 10.138.0.19
Thu Oct 29 00:21:18 UTC All agents have connected!                                                                                                                                            
^CPress Ctrl+C again to abort the installation. 
^CReceived interrupt signal. Aborting the installer gracefully, please wait.
[ERROR]: operation aborted
```

Try to install 7.0.23 the same way:

```
-bash-4.2$ sudo /home/centos/7.0.23/gravity install --cloud-provider=generic                    
Thu Oct 29 00:21:48 UTC Starting enterprise installer
[ERROR]: failed to validate installer directory. Make sure you're running the installer from the directory with the contents of the installer tarball
        specified state directory /home/centos/test does not contain application data, please provide a path to the unpacked installer tarball or specify an application package via --app flag
```
### Environment

 - Gravity version [e.g. 7.0.11]: 7.0.23
 - OS [e.g. Redhat 7.4]: CentOS 7.8
 - Platform [e.g. Vmware, AWS]: GCP

**Comment by wadells** [comment]
Workaround, supply the installer directory as a positional argument:

```
-bash-4.2$ sudo /home/centos/7.0.23/gravity install --cloud-provider=generic /home/centos/7.0.23
Thu Oct 29 00:29:38 UTC Starting enterprise installer

To abort the installation and clean up the system,
press Ctrl+C two times in a row.

If you get disconnected from the terminal, you can reconnect to the installer
agent by issuing 'gravity resume' command.

If the installation fails, use 'gravity plan' to inspect the state and
'gravity resume' to continue the operation.
See https://gravitational.com/gravity/docs/cluster/#managing-an-ongoing-operation for details.

Thu Oct 29 00:29:38 UTC Connecting to installer
Thu Oct 29 00:29:45 UTC Connected to installer
Thu Oct 29 00:29:46 UTC Successfully added "node" node on 10.138.0.19
Thu Oct 29 00:29:46 UTC All agents have connected!
^CPress Ctrl+C again to abort the installation.
^CReceived interrupt signal. Aborting the installer gracefully, please wait.
[ERROR]: operation aborted
```

**Comment by wadells** [comment]
A bisect shows this was introduced in 7.0.22.  Likely via https://github.com/gravitational/gravity/pull/2240.

It is not present in master (7.1 alpha):

```
-bash-4.2$ sudo /home/centos/7.1.0-alpha.1.250/gravity install --cloud-provider=generic
Thu Oct 29 00:50:34 UTC Starting enterprise installer

To abort the installation and clean up the system,
press Ctrl+C two times in a row.

If you get disconnected from the terminal, you can reconnect to the installer
agent by issuing 'gravity resume' command.

If the installation fails, use 'gravity plan' to inspect the state and
'gravity resume' to continue the operation.
See https://gravitational.com/gravity/docs/cluster/#managing-operations for details.

Thu Oct 29 00:50:35 UTC Connecting to installer
Thu Oct 29 00:50:43 UTC Connected to installer
Thu Oct 29 00:50:44 UTC Successfully added "node" node on 10.138.0.19
Thu Oct 29 00:50:44 UTC All agents have connected!
Thu Oct 29 00:50:44 UTC Operation has been created
Thu Oct 29 00:50:46 UTC Executing "/init/walt-70-upgrade-node-0" locally
Thu Oct 29 00:50:46 UTC Initialize operation on node walt-70-upgrade-node-0
```

cc @a-palchikov

**Comment by wadells** [comment]
Fixed in 7.0.24 by https://github.com/gravitational/gravity/pull/2295.
